### PR TITLE
docs (Sessions): Commenting Functions

### PR DESCRIPTION
**Pull request by WispyMouse**
This set of changes is just comments on the functions and regions of the code. Minimal amounts of reorganization or renaming has been done. There should be no functional changes.

There's lots of TODOs to remind myself of what reworks are needed for each function. This was a useful exercise for determining things that I think need to change at a glance, and was a great opportunity to take inventory.

I would like feedback on the commenting style and details, if anyone has any. I'll merge this over by midday tomorrow.

### CODE REVIEW

File: Assets/Scripts/EOSSessionsManager.cs [hunk 1]
```diff
@@ -348,17 +351,40 @@ public class EOSSessionsManager : IEOSSubManager
         private SessionDetails JoiningSessionDetails = null;
         private ulong JoinedSessionIndex = 0;
 
+        /// <summary>
+        /// When joining a Session you did not create, the Session's name is not shared with you.
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs]: nit: It would be clearer to say simply, "...the Session name..."
I would also rephrase to avoid use of any pronouns (ie "you").

File: Assets/Scripts/EOSSessionsManager.cs [hunk 2]
```diff
@@ -348,17 +351,40 @@ public class EOSSessionsManager : IEOSSubManager
...
+        /// Instead the Session is given a local name, starting with this, when <see cref="GenerateJoinedSessionName(bool)"/> is called.
+        /// This local name doesn't have to follow any particular formatting, but does need to be locally unique and consistent.
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs]: nit: Actually, as a matter of habit I try to avoid using contractions at all. However, sometimes it is unavoidable.

File: Assets/Scripts/EOSSessionsManager.cs [hunk 3]
```diff
@@ -348,17 +351,40 @@ public class EOSSessionsManager : IEOSSubManager
...
         private const string BUCKET_ID = "SessionSample:Region";
 
+        /// <summary>
+        /// When creating a Session or searching for one, this is the attribute used to filter for buckets.
+        /// In these sample functions, <see cref="CreateSession(Session, bool, Action)"/> sets the BucketId
+        /// in <see cref="CreateSessionModificationOptions"/> as well as an <see cref="SessionAttribute"/>.
+        /// </summary>
         private const string EOS_SESSIONS_SEARCH_BUCKET_ID = "bucket";
+
+        // TODO: All three of these are unused, remove?
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs]: nit: I would categorize this as a `NOTE:`, the `TODO:` version would be "Confirm that the three proceeding values are unused, and if so remove them."

File: Assets/Scripts/EOSSessionsManager.cs [hunk 4]
```diff
@@ -348,17 +351,40 @@ public class EOSSessionsManager : IEOSSubManager
...
         private const string EOS_SESSIONS_SEARCH_EMPTY_SERVERS_ONLY = "emptyonly";
         private const string EOS_SESSIONS_SEARCH_NONEMPTY_SERVERS_ONLY = "nonemptyonly";
         private const string EOS_SESSIONS_SEARCH_MINSLOTSAVAILABLE = "minslotsavailable";
 
         // UI Parameterized Callbacks
+        // TODO: None of these are subscribed to, several of them aren't implemented fully. Remove? Implement more fully?
+        // UIOnSessionCreated is actually utilized
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs]: I'd be more specific with respect to which you mean when you say, "these," particularly since you follow up to specify one in particular. Simpler would be to just limit the note to specifically reference the ones it applies to, or be more verbose.

File: Assets/Scripts/EOSSessionsManager.cs [hunk 5]
```diff
@@ -377,6 +403,12 @@ public class EOSSessionsManager : IEOSSubManager
 
         #region Peer2Peer Messaging Variables
 
+        /* The Sessions Interface does not provide a way to notify Session Owners and Members when things change,
+         * such as users joining, leaving, or the session starting, ending, being modified, or bieng destroyed.
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs]: spelling: bieng = being

File: Assets/Scripts/EOSSessionsManager.cs [hunk 6]
```diff
@@ -458,14 +494,21 @@ public class EOSSessionsManager : IEOSSubManager
 
         /// <summary>
         /// After a successful <see cref="RefreshSession(string)"/> and <see cref="OnRefreshSessionFindSessionsCompleteCallback(ref SessionSearchFindCallbackInfo)"/> callback call,
-        /// this is run to inform UI of the updated Session information.
+        /// this is run to inform UI of the updated Session information. This should be subscribed to by your program's UI.
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs]: A bit redundant. Did you intend to replace the comment instead of amending it? Either way, capitalization.

File: Assets/Scripts/EOSSessionsManager.cs [hunk 7]
```diff
@@ -458,14 +494,21 @@ public class EOSSessionsManager : IEOSSubManager
...
         /// </summary>
         public Action<Session, SessionDetails> UIOnSessionRefresh { get; set; }
 
         #endregion
 
         #region Manager Set Up And Utility
 
+        /* This region contains meta-management for this Manager class.
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs]: For multi-line block comments, always start the block with an empty comment line, and end with an empty comment line:

```csharp
/*
 * REGION NOTES:
 * 
 * This would be an example (the header of "REGION NOTES" isn't prescriptive, 
 * but can be a useful way to indicate the intent of the comment block that is
 * not otherwise adjacent to a function or class definition).
 */
```

File: Assets/Scripts/EOSSessionsManager.cs [hunk 8]
```diff
@@ -494,6 +537,11 @@ internal static void Log(string toPrint)
             UnityEngine.Debug.Log(toPrint);
         }
 
+        /// <summary>
+        /// Checks for new Peer2Peer messages for Session management, as well as updating the OnlineSessionState of ActiveSessions.
+        /// This Manager is not a MonoBehaviour, so this function should be run by the program every update.
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs:553]: The comment is correct, and this is certainly outside the scope of the task of adding comments to existing code - but it brings up a point that might otherwise not have been noticed: is it possible that `EOSSessionsManager` _should_ be a `MonoBehaviour` (which would obviate the need to make this comment)?

File: Assets/Scripts/EOSSessionsManager.cs [hunk 9]
```diff
@@ -544,14 +596,20 @@ public void OnLoggedIn()
...
 
             CurrentInvite = null;
 
             userLoggedIn = true;
         }
 
+        /// <summary>
+        /// Unsubscribes from invitations and Peer2Peer message requests and clears internal states.
+        /// This can be run before or after a successful logout, or preparing to shut down the application.
+        /// This must be called in order to remove Native notifications.
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs]: It may be wise to be clearer regarding what Native notifications means in this context.

File: Assets/Scripts/EOSSessionsManager.cs [hunk 10]
```diff
@@ -574,13 +632,31 @@ public void OnLoggedOut()
             userLoggedIn = false;
         }
 
+        /// <summary>
+        /// Unsubscribes from invitations and Peer2Peer message requests.
+        /// This should be called before the application finishes shutting down in order to properly remove Native notifications.
+        /// TODO: This function has no references, and is incomplete in its implementation. This will likely be removed.
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs:649]: I think more properly it _should_ be called, as (just going based off of the contents of the function) it seems to be freeing some resources. I have noticed that this seems to be the case with several of the managers.

File: Assets/Scripts/EOSSessionsManager.cs [hunk 11]
```diff
@@ -574,13 +632,31 @@ public void OnLoggedOut()
...
+        /// </summary>
+        [Obsolete]
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs:651]: Neat! I didn't know this was something that could be done. Thanks for teaching me something new! 😀

File: Assets/Scripts/EOSSessionsManager.cs [hunk 12]
```diff
@@ -574,13 +632,31 @@ public void OnLoggedOut()
...
             UnsubscribeToSessionMessageConnectionRequests();
         }
 
+        /// <summary>
+        /// Informs the <see cref="UIInterface"/> that a UI Event has been processed.
+        /// Some functions in the EOS SDK return a UIEventId. When this happens, it is required that this function be called to released by the SDK.
+        /// For example, if you are subscribed to <see cref="OnJoinSessionAcceptedListener(ref JoinSessionAcceptedCallbackInfo)"/>, after a successful call to <see cref="SessionsInterface.JoinSession(ref JoinSessionOptions, object, OnJoinSessionCallback)"/>,
+        /// the notification function will receive a <see cref="JoinSessionAcceptedCallbackInfo.UiEventId"/>.
+        /// You can use this event id to create a <see cref="SessionDetails"/> object by calling <see cref="MakeSessionHandleByEventId(ulong)"/>.
+        /// After you've done so, or if you aren't going to use this handle, you must call this function to release the handle in the EOS SDK.
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs]: An example of how to rephrase to remove pronouns:

```csharp
/// For example, if a subscription is made to <see cref="OnJoinSessionAcceptedListener(ref JoinSessionAcceptedCallbackInfo)"/>, after a successful call to <see cref="SessionsInterface.JoinSession(ref JoinSessionOptions, object, onJoinSessionCallback)"/>, 
/// the notification function will receive a <see cref="JoinSessionAcceptedCallbackInfo.UiEventId"/>.
/// This event id can be used to create a <see cref="SessionDetails"/> object by calling <see cref="MakeSessionHandleByEventId(ulong)"/>.
/// After doing so, this function must be called to release the handle within the EOS SDK.
```

File: Assets/Scripts/EOSSessionsManager.cs [hunk 13]
```diff
@@ -574,13 +632,31 @@ public void OnLoggedOut()
...
+        /// Most of the functions in the EOS SDK that require this will mention it in the comments of the callback function. Anything with a UiEventId should have this called.
+        /// 
+        /// TODO: This is hard coded to only handle the <see cref="JoinUiEvent"/> case. It must be changed to also accept the event's id in order to function properly.
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs:668]: nit: Another example where the contraction can be removed without altering the meaning of the comment.

File: Assets/Scripts/EOSSessionsManager.cs [hunk 14]
```diff
@@ -596,11 +672,25 @@ private void AcknowledgeEventId(Result result)
...
+        /// Sets the current user's Social Presence.
+        /// 
+        /// TODO: This is not called by this manager. It either should be utilized, or moved to another manager, perhaps some sort of EOSPresenceManager.
+        /// </summary>
+        /// <param name="joinInfo">If provided, this string will be displayed to users in the Social Overlay as your current presence.
+        /// This can be used to indicate things like current level, looking for party status, etc.</param>
+        /// <param name="onLoggingOut">Indicates if the user is currently attempting or has already logged out.
+        /// If true, then if there's an error while trying to change your presence, the error will be ignored.
+        /// 
+        /// TODO: The error handling in both scenarios is almost identical. Perhaps an error state should be returned in the situation where this is false and an error is encountered.</param>
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs]: If you choose to have line breaks within your comment tags, you should put the open and close tags on their own lines.

File: Assets/Scripts/EOSSessionsManager.cs [hunk 15]
```diff
@@ -965,6 +1151,17 @@ public SessionDetails MakeSessionHandleFromSearch(string sessionId)
 
         #region Online Session Lookup and Search
 
+        /// <summary>
+        /// Performs an online Search to look up a Session, where all attributes provided are met.
+        /// <see cref="OnFindSessionsCompleteCallback"/> is called with the results.
+        /// A handle for retriving results will be stored in <see cref="CurrentSearch"/>.
+        /// 
+        /// TODO: I feel like this should run a callback or return a Result in case of errors when trying to set up the search.
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs]: Particularly for comments that contain personal statements or opinions, you should either indicate that it is a `NOTE:` to yourself or rephrase the comment such that it is more objective. As it is written, it is not really a `TODO:` item that is written for any developer to take action on.

File: Assets/Scripts/EOSSessionsManager.cs [hunk 16]
```diff
@@ -965,6 +1151,17 @@ public SessionDetails MakeSessionHandleFromSearch(string sessionId)
...
+        /// TODO: Only finds up to 10 results. Should have some method for finding next sets of results.
```
Comment (paulhazen) [inline Assets/Scripts/EOSSessionsManager.cs:1181]: This is a great example of what I think of as an appropriate `TODO:` comment. It clearly states what needs to be changed, and gives context for why.

**Review by paulhazen — APPROVED** [review]
I ran out of time to go through everything, apologies for being incomplete. Any comment I left that starts with a `nit:` you can feel free to take it or leave it and mark the conversation as resolved once you decide what to do with the feedback.

I'm approving, but I _do_ have some things I would change in the comments if this was being merged into `development`, so I'm depending on the requirement that conversations be resolved prior to merging to serve as a gating mechanism.

File: Assets/Scripts/EOSSessionsManager.cs (hunk 1 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs]: Noted! I am definitely awkward with pronouns in these comments, and it seems really easy to rephrase those away.

File: Assets/Scripts/EOSSessionsManager.cs (hunk 2 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs]: "doesn't" easily becomes "does not". But "Session's" is a possessive, is that still a contraction?

File: Assets/Scripts/EOSSessionsManager.cs (hunk 3 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs]: This is definitely meant to be followed up on, so I'll rephrase this to be more like a TODO comment.

File: Assets/Scripts/EOSSessionsManager.cs (hunk 4 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs]: Specified. I decided the clearest way to do that was to put `UIOnSessionCreated` in to its own singular spot, make a line break, and then comment about the following being unimplemented.

File: Assets/Scripts/EOSSessionsManager.cs (hunk 5 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs]: Good catch, typo fixed.

File: Assets/Scripts/EOSSessionsManager.cs (hunk 6 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs]: I don't see the capitalization fault, though I do see a "you/your" pronoun usage to replace.

I want a clear way to communicate, HEY USER! Subscribe to this or you're missing out!

File: Assets/Scripts/EOSSessionsManager.cs (hunk 7 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs]: Sure, that's a simple pattern to implement. I'll put that in all the regional comments.

File: Assets/Scripts/EOSSessionsManager.cs (hunk 8 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs:553]: Mayyyyyyybe? I don't think it should be, because none of the other Manager classes from the samples are MonoBehaviours. Unless it is okay to break that pattern? Either way it would change how this particular manager needs to be implemented.

File: Assets/Scripts/EOSSessionsManager.cs (hunk 9 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs]: Sure, I think the easiest way to do that here is just to point at OnLoggedIn, so changed the comments for that.

File: Assets/Scripts/EOSSessionsManager.cs (hunk 10 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs:649]: This already uses the word "should". Would this be better as "must"?

File: Assets/Scripts/EOSSessionsManager.cs (hunk 11 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs:651]: Handy tag is handy! ^o^ You can also use it to point users towards the new function that should be called, in the argument for this attribute.

File: Assets/Scripts/EOSSessionsManager.cs (hunk 12 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs]: Seems like a clear rewrite, I'll use most of your text. I do think the "or if you aren't going to use this handle" is important, so I preserved it in the form of "or if the handle will not be utilized". Otherwise I think it implies "after using this call Acknowledge" rather than what is really intended of "after it's not useful then call Acknowledge".

File: Assets/Scripts/EOSSessionsManager.cs (hunk 13 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs:668]: This contraction got eliminated with the rephrase, but I'll note to use "you have" if this ever comes up again.

File: Assets/Scripts/EOSSessionsManager.cs (hunk 14 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs]: That seems like a reasonable and clean guideline, will implement that.

File: Assets/Scripts/EOSSessionsManager.cs (hunk 15 above)
Comment (WispyMouse) [reply Assets/Scripts/EOSSessionsManager.cs]: I'm willing to put on my executive action pants on, and restate this to "this should be done" rather than "I feel like this should".

**Comment by WispyMouse** [comment]
All "you"-s have been replaced with language about "the local user" or "the application".
